src/testmenu.cpp

```diff
@@ -0,0 +1,778 @@
+                }
+            }
+            if (TestMenu::getMenuState() == testMenuState_t::CLEAR)
+            {
+                _currentMenu->display();
+            }
+            return;
+        }
+    }
+    Serial.println(F("Invalid key pressed."));
+
+    _currentMenu->display();
+}
+
+void TestMenu::displayStepperPos() const
+{
+    String statusRaDec;
+    statusRaDec = F("  RA: ");
+    statusRaDec += rightJustify(String(mount.getCurrentStepperPosition(RA_STEPS)), 8);
+    statusRaDec += mount.isAxisRunning(RA_STEPS) ? "^" : " ";
+    statusRaDec += F("   ALT: ");
+    #if (ALT_STEPPER_TYPE != STEPPER_TYPE_NONE)
+    String altSteps = "n/a";
+    #else
+    String altSteps = String(mount.getCurrentStepperPosition(ALTITUDE_STEPS));
+    #endif
+    statusRaDec += rightJustify(altSteps, 8);
+    statusRaDec += mount.isAxisRunning(ALTITUDE_STEPS) ? "^" : " ";
+    statusRaDec += F("   TRK: ");
+    statusRaDec += rightJustify(String(mount.getCurrentStepperPosition(TRACKING)), 8);
+    statusRaDec += mount.isSlewingTRK() ? "^" : " ";
+
+    String statusAltAz;
+    statusAltAz = F(" DEC: ");
+    statusAltAz += rightJustify(String(mount.getCurrentStepperPosition(DEC_STEPS)), 8);
+    statusAltAz += mount.isAxisRunning(DEC_STEPS) ? "^" : " ";
+    statusAltAz += F("    AZ: ");
+    #if (ALT_STEPPER_TYPE != STEPPER_TYPE_NONE)
+    String azSteps = "n/a";
+    #else
+    String azSteps  = String(mount.getCurrentStepperPosition(AZIMUTH_STEPS));
+    #endif
+    statusAltAz += rightJustify(azSteps, 8);
+    statusAltAz += mount.isAxisRunning(AZIMUTH_STEPS) ? "^" : " ";
+    statusAltAz += F("   FOC: ");
+    #if (ALT_STEPPER_TYPE != STEPPER_TYPE_NONE)
+    String focSteps = "n/a";
+    #else
+    String focSteps = String(mount.getCurrentStepperPosition(FOCUS_STEPS));
+    #endif
+    statusAltAz += rightJustify(focSteps, 8);
+    statusAltAz += mount.isAxisRunning(FOCUS_STEPS) ? "^" : " ";
+    Serial.println(statusRaDec);
+    Serial.println(statusAltAz);
+}
+
+void TestMenu::display() const
+{
+    Serial.println("");
+
+    if (_level == 0)
+    {
+        Serial.println(F("**************************************"));
+    #ifdef OAM
+        Serial.println(F("*** OpenAstroMount (OAM) Test Menu ***"));
+    #else
+        Serial.println(F("** OpenAstroTracker (OAT) Test Menu **"));
+    #endif
+        Serial.print(F("************* "));
+        Serial.print(freeMemory());
+        Serial.println(F(" bytes *************"));
+        displayStepperPos();
+        Serial.println(F("**************************************"));
+    }
+    else
+    {
+        Serial.print(F("---------------- "));
+        Serial.print(freeMemory());
+        Serial.println(F(" bytes ------------"));
+        displayStepperPos();
+        Serial.println(F("-----------------------------------------"));
+        Serial.print("  ");
+        Serial.print(_name);
+        Serial.println(F(" Menu"));
+        Serial.println(F("--------------------------"));
```

Comment:
You should make these `-` strings the same length (or group them into same lengths) to reduce memory usage (they get deduplicated)

Reply:
Since they are stored in program (Flash memory), I'd rather have the output looking nice, than saving a few bytes of program memory. If it was stack memory, sure.

Reply:
Yeah I'm just saying within reason, like for example line 660 & 664 should be the same length

Reply:
I mean they are that length so they line up. If I change it, it'll look like all raggedy

Reply:
Ok no problem then. I don't know how it's supposed to look, I was just pointing out the option